### PR TITLE
(DOCSP-28276) Add argument to `useQuery` to allow result to be filtered and sorted

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

This PR updates the syntax for `useQuery()` when it's used with `.filtered()` and `.sorted()`. Essentially, going from this:
```JS
const contactsInArea = contacts.filtered(`address.postalCode == '${postalCode}'`);
```
to this:
```JS
const contactsInArea = useQuery(
  Contact,
  contacts => {
    return contacts.filtered(`address.postalCode == '${postalCode}'`);
  },
  [postalCode],
);
```

The new syntax has baked-in performance improvements and is the recommended way to handle filtering and sorted. We should avoid `useQuery(<ObjectClass>).sorted()` or `useQuery(<ObjectClass>).filtered()`.

### Jira

- https://jira.mongodb.org/browse/DOCSP-28276

### Staged Changes

- [Code samples throughout the React Native SDK docs](https://preview-mongodbkrollinsmdb.gatsbyjs.io/realm/DOCSP-28276/sdk/react-native/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Comment by krollins-mdb** [comment]
I'm not sure what the test error is. I'm going to spend time this week fixing tests and the test suite.

File: source/examples/generated/react-native/js/dictionary-test.snippet.query-objects-with-dictionary.jsx [hunk 1]
```diff
@@ -4,17 +4,22 @@ const HomeList = () => {
...
+    return homeOwners.filtered('home.@keys = "price"');
+  });
 
   // run the `.filtered()` method on all the returned homeOwners to
   // find the house with the address "Summerhill St."
-  const summerHillHouse = homeOwners.filtered(
-    'home["address"] = "Summerhill St."',
-  )[0].home;
+  const summerHillHouse = useQuery(HomeOwner, homeOwners => {
+    return homeOwner.filtered('home["address"] = "Summerhill St."');
```
Comment (lindseymoore) [inline source/examples/generated/react-native/js/dictionary-test.snippet.query-objects-with-dictionary.jsx:14]:
```suggestion
    return homeOwners.filtered('home["address"] = "Summerhill St."');
```
Is this supposed to be ``homeOwners``?

**Review by lindseymoore — CHANGES_REQUESTED** [review]
1 question!

**Review by lindseymoore — APPROVED** [review]
Going to go ahead and approve since I'll be OOO. Please review previous question!

File: source/examples/generated/react-native/js/dictionary-test.snippet.query-objects-with-dictionary.jsx (hunk 1 above)
Comment (krollins-mdb) [reply source/examples/generated/react-native/js/dictionary-test.snippet.query-objects-with-dictionary.jsx:14]: Yep! Thanks for catching that.

**Pull request merged**